### ISSUE TITLE
[Feature]: data source for Directus / NocoDB / Grist

### ISSUE THREAD

**Issue by tiger360** [appsmithorg/appsmith#11656]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

To build an integration of Directus / NocoDB / Grist with their OpenAPI Specification 
https://docs.directus.io/reference/system/server/#get-openapi-specification

### Why should this be worked on?

To make it smoother like on n8n when we need to bind the datasource with the rest api .
And also I was browsing in a subreddit "selfhosted" and I saw Budibase asking to build data source for Grist too.

**Comment by MyWay** [comment]
I agree that these integrations would make a difference among these tools in terms of productivity and ux. See tooljet with appwrite for instance.
